codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/shapevisitor/DafnyToAwsSdkShapeVisitor.java

```diff
@@ -377,86 +370,54 @@ public String booleanShape(final BooleanShape shape) {
   public String stringShape(final StringShape shape) {
     writer.addImportFromModule(DAFNY_RUNTIME_GO_LIBRARY_MODULE, "dafny");
     if (shape.hasTrait(EnumTrait.class)) {
+      var nilCheck = "";
       if (this.isOptional) {
-        final String unAssertedDataSource = dataSource.startsWith("input.(")
+        var unAssertedDataSource = dataSource.startsWith("input.(")
```

Comment:
Do we reassign this anywhere?

Reply:
It was but then I ended up simplifying a lot of if blocks - finalized it.